modules/save_contigs_per_kingdom.nf

```diff
@@ -3,15 +3,16 @@ process SAVE_CONTIGS_PER_KINGDOM {
 
   input:
   val contigsPath
+  val index
   each result
 
   output: 
-  tuple val("${result[4]}"), stdout
+  tuple val("${result[index]}"), path("output.save_contigs_per_kingdom")
 
   script:
 
   """
-  grep -Pzo '>${result[0]}\\s[^\\>]*' $contigsPath | sed '\$ s/.\$//'
+  grep -Pzo '>${result[0]}\\s[^\\>]*' $contigsPath | sed '\$ s/.\$//' > output.save_contigs_per_kingdom
```

Comment:
If I understand this correctly you're saving fasta's per first prediction. Since you have the TSV with all the prediction information I think it is enough to have:
- an intermediate fasta for CAT confirmiation (not a final output file)
- two final output files eukaryotes.fasta other_kingdoms.fasta or something similar.

Reply:
I'm not sure this is what you're talking about, but the current script already save fastas during the first step as firstStep_contigs[firstPredictionClass].fa and, for non-sure prediction, as firstStep_RequiringCATvalidation.fa, and then after the second step as secondStep_Eukaryotes.fa and secondStep_NotEukaryotes.fa. 

So if I've understood your message correctly, that's already the case ! :)
I just need to change the fasta names, which aren't very pretty.